TEST_APPS/testcases/nfc/test_nfc.py

```diff
@@ -0,0 +1,316 @@
+    response = self.nfc_command("dev1", "iseeprom")
+    eeprom = response.parsed['iseeprom']
+
+    self.logger.info("Target includes NFCEEPROM: %s" % eeprom)
+    if not eeprom:
+        self.nfc_command("dev1", "start")
+
+    # write poster tag to target
+    self.command("dev1", "setsmartposter %s" % expectedURI)
+
+    self.clf.nfc.connect()
+    tag = self.clf.clf_response()
+    asserts.assertNotNone(tag, "Could not connect to any tag")
+    asserts.assertEqual(1, len(tag.ndef.records), "expected number NDEF records")
+
+    asserts.assertEqual(tag.ndef.records[0].__class__.__name__, "SmartposterRecord", "expected SmartposterRecord")
+    asserts.assertEqual(expectedURI, tag.ndef.records[0].uri_records[0].uri, "expected exact URI")
+
+
+@test_case(CreamSconeTests)
+def test_nfce2e_reprogrammed(self):
+    """
+    check - Tag can be programmed from a remote and read via contactless
+    """
+    expectedURI = "https://www.google.com"
+
+    response = self.nfc_command("dev1", "iseeprom")
+    eeprom = response.parsed['iseeprom']
+    self.logger.info("Target includes NFCEEPROM: %s" % eeprom)
+
+    self.nfc_command("dev1", "initnfc")
+    if not eeprom:
+        self.nfc_command("dev1", "start")
+    self.nfc_command("dev1", "erase")
+
+    # program a poster tag to target
+    print("Write Smartposter MESSAGE wirelessly")
+    self.clf.nfc.connect()
+    tag = self.clf.clf_response()
+    asserts.assertNotNone(tag, "Could not connect to any tag")
+    smartposter = nfc_messages.make_smartposter(expectedURI, ["en-US:Other search engines exist"])
+    nfc_messages.program_remote_tag(smartposter, tag)
+    self.logger.info("Remote programmed %d bytes Smartposter" % len(str(smartposter)))
+
+    print("Write back Smartposter MESSAGE wirelessly")
+    self.clf.nfc.connect()
+    tag = self.clf.clf_response()
+    asserts.assertNotNone(tag, "Could not re-connect to any tag")
+
+    asserts.assertEqual(tag.ndef.records[0].__class__.__name__, "SmartposterRecord", "expected SmartposterRecord")
+    asserts.assertEqual(expectedURI, tag.ndef.records[0].uri_records[0].uri, "expected exact URI")
+    self.clf.nfc.mute()    # disable radio, to allow a local session
+
+    # verify in target
+    response = self.nfc_command("dev1", "readmessage")
+
+    # check contents
+    expected_message = str(smartposter)
+    self.assert_binary_equal(response.parsed['nfcmessage'], expected_message)
+
+
+@test_case(CreamSconeTests)
+def test_nfce2e_read_stress(self):
+    """
+    check - Large record can be programmed in and read via contactless
+    """
+    messageRep = 'thequickbrownfoxjumpedoverthelazydog' # repeating message written
+
+    response = self.nfc_command("dev1", "iseeprom")
+    eeprom = response.parsed['iseeprom']
+    self.logger.info("Target includes NFCEEPROM: %s" % eeprom)
+    max_ndef = STRESS_BUFFLEN
+
+    self.nfc_command("dev1", "initnfc")
+    if not eeprom:
+        self.nfc_command("dev1", "start")
```

Comment:
so this tests is for both wireless and testing and testing writing eeprom locally? all the comments appear to indicate otherwise